packages/test-sync/src/commonTest/kotlin/io/realm/kotlin/test/mongodb/common/GeoSpatialTests.kt

```diff
@@ -0,0 +1,229 @@
+package io.realm.kotlin.test.mongodb.common
+
+import io.realm.kotlin.Realm
+import io.realm.kotlin.annotations.ExperimentalGeoSpatialApi
+import io.realm.kotlin.entities.Location
+import io.realm.kotlin.entities.sync.SyncRestaurant
+import io.realm.kotlin.ext.query
+import io.realm.kotlin.internal.platform.runBlocking
+import io.realm.kotlin.log.LogLevel
+import io.realm.kotlin.mongodb.User
+import io.realm.kotlin.mongodb.subscriptions
+import io.realm.kotlin.mongodb.sync.SyncConfiguration
+import io.realm.kotlin.mongodb.syncSession
+import io.realm.kotlin.test.mongodb.TEST_APP_FLEX
+import io.realm.kotlin.test.mongodb.TestApp
+import io.realm.kotlin.test.mongodb.createUserAndLogIn
+import io.realm.kotlin.test.util.TestHelper
+import io.realm.kotlin.test.util.use
+import io.realm.kotlin.types.geo.Distance
+import io.realm.kotlin.types.geo.GeoBox
+import io.realm.kotlin.types.geo.GeoCircle
+import io.realm.kotlin.types.geo.GeoPoint
+import io.realm.kotlin.types.geo.GeoPolygon
+import kotlinx.coroutines.delay
+import org.mongodb.kbson.ObjectId
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFails
+import kotlin.time.Duration.Companion.seconds
+
+private val GEO_SCHEMA = setOf(SyncRestaurant::class, Location::class)
```

Comment:
We cannot use this anymore. All schemas need to be the same after the server introduced the new developer mode. I have a branch you can work from.

Reply:
See https://github.com/realm/realm-kotlin/pull/1535